extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddNamespaceDecorator.java

```diff
@@ -1,27 +1,39 @@
 package io.quarkus.kubernetes.deployment;
 
+import java.util.List;
 import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
 
-import io.dekorate.kubernetes.decorator.AddSidecarDecorator;
 import io.dekorate.kubernetes.decorator.Decorator;
-import io.dekorate.kubernetes.decorator.NamedResourceDecorator;
 import io.dekorate.kubernetes.decorator.ResourceProvidingDecorator;
-import io.fabric8.kubernetes.api.model.ObjectMeta;
-import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
 
-public class AddNamespaceDecorator extends NamedResourceDecorator<ObjectMetaBuilder> {
+public class AddNamespaceDecorator extends Decorator<KubernetesListBuilder> {
 
+    private static final Set<String> CLUSTERED_RESOURCES = Set.of("ClusterRoleBinding", "ClusterRole");
     private final String namespace;
 
     public AddNamespaceDecorator(String namespace) {
         this.namespace = Objects.requireNonNull(namespace);
     }
 
     @Override
-    public void andThenVisit(ObjectMetaBuilder builder, ObjectMeta resourceMeta) {
-        if (!builder.hasNamespace()) {
-            builder.withNamespace(namespace);
-        }
+    public void visit(KubernetesListBuilder list) {
+        List<HasMetadata> buildItems = list.buildItems()
+                .stream()
+                .filter(obj -> obj instanceof HasMetadata)
```

Comment:
At the very least, you don't need this: all the items in the list should be `HasMetadata` instances.